pkg/model/model.go

```diff
@@ -0,0 +1,282 @@
+	if position < 1 || position > 128 {
+		err = errors.New("Failed partitioning posititon: " + strconv.FormatUint(uint64(position), 10) + " A partition position must be >= 1 && <= 128")
+		return
+	}
+
+	p = &Partition{
+		Name:     name,
+		Position: position,
+		Size:     size,
+		Type:     ptype,
+	}
+
+	return
+}
+
+func NewBlockDevice(file string) (bd *BlockDevice, err error) {
+	bd = &BlockDevice{
+		File: file,
+	}
+
+	if !bd.Validate() {
+		err = errors.New("Block Device " + bd.File + " failed validation.")
+	}
+
+	return
+}
+
+// NewBlockDevices returns a slice of BlockDevice(s) for the supplied
+// slice of strings listing device files.
+func NewBlockDevices(s_devices ...string) (block_devices []*BlockDevice, err error) {
+	for _, dev := range s_devices {
+		bd, bd_err := NewBlockDevice(dev)
+		if err != nil {
+			return block_devices, bd_err
+		}
+
+		block_devices = append(block_devices, bd)
+	}
+
+	return
+}
+
+// GetDeviceFiles returns a slice of strings with all the device files
+// that make up the given RaidArray.
+// It returns a slice of strings and possibly an error.
+func (a RaidArray) GetDeviceFiles() (device_files []string, err error) {
+	for _, dev := range a.Devices {
+		device_files = append(device_files, dev.File)
+	}
+
+	return
+}
+
+// ValidateDevices validates that each block device is 'valid' by calling
+// Validate on each BlockDevice.
+// It returns false if any of the underlying calls to Validate() are false.
+func (a RaidArray) ValidateDevices() (valid bool) {
+	for _, bd := range a.Devices {
+		if !bd.Validate() {
+			return false
+		}
+	}
+
+	return true
+}
+
+// Validate validates that the given block device is correct and accessible.
+// It returns an bool indicating pass/fail.
+func (b BlockDevice) Validate() bool {
+	resolved_path, err := filepath.EvalSymlinks(b.File)
+
+	if err != nil {
+		return false
+	}
+
+	fi, err := os.Stat(resolved_path) // Returns err if file is not accessible
+
+	if os.IsNotExist(err) {
+		return false
+	}
+
+	m := fi.Mode()
+
+	return m&os.ModeDevice != 0
+}
+
+// Format prepares a Partition on a given BlockDevice with a file system
+// It returns an error object, or nil depending on the results.
+func (p Partition) Format() (out string, err error) {
+	switch f := p.FileSystem; f {
+	case "swap":
+		out, err = CallCommand("mkswap", p.BlockDevice.File)
+	default:
+		// TODO(jwb) Check for the existence of mkfs.FileSystem here
+		mkfs_options := []string{"-F"}
+		mkfs_options = append(mkfs_options, p.FileSystemOptions...)
+		mkfs_options = append(mkfs_options, p.BlockDevice.File)
+		out, err = CallCommand("mkfs."+p.FileSystem, mkfs_options...)
+	}
+
+	return
+}
+
+func (p Partition) GetUUID() (string, error) {
+	b_uuid, err := CallCommand("blkid", "-s", "UUID", "-o", "value", p.BlockDevice.File)
+	return strings.TrimRight(string(b_uuid), "\n"), err
+}
+
+func CallCommand(cmd_name string, cmd_options ...string) (out string, err error) {
```

Comment:
suggestion: move this method and any other similar ones into its own file, since its more of helper method and isn't a method on a struct.

Reply:
Pass `context.Context` to `CallCommand` and use `exec.CommandContext` instead of `exec.Command` https://pkg.go.dev/os/exec#CommandContext.

See my other comment on context passing.

Reply:
moved callCommand into internal/command/command.go package 'command' and changed the fun name to simply Call